### PR TITLE
Upgrade ember cli

### LINKED ISSUE #361: Does this addon still need the "ember-runtime-enumerable-includes-polyfill" dependency?
The "ember-runtime-enumerable-includes-polyfill" repository has been archived by the owner, see [repository](https://github.com/ember-polyfills/ember-runtime-enumerable-includes-polyfill).

Does this addon still needs it?

We saw this when updating to "ember-source": "3.27.5"
We use "ember-cli-flash": "2.1.3"

```
WARNING: [DEPRECATION] [DEPRECATION] Usage of the Ember Global is deprecated. You should import the Ember module or the specific API instead.

See https://deprecations.emberjs.com/v3.x/#toc_ember-global for details.

Usages of the Ember Global may be caused by an outdated ember-cli-babel dependency.

The following outdated versions are found in your project:

* ember-cli-babel@6.18.0, currently used by:
  * ember-runtime-enumerable-includes-polyfill@2.1.0 (Dormant)
    * Depends on ember-cli-babel@^6.9.0
    * Added by ember-cli-flash@2.1.3
```

### PR DESCRIPTION
**Pull request by stukalin**
Fix #361

### CODE REVIEW

**Comment by stukalin** [comment]
Note that's the canary build which causes the tests failure. In theory we can temporarily disable it. What do you think?

**Comment by stukalin** [comment]
@poteto may I ping you on this?

**Comment by ndekeister-us** [comment]
@stukalin issue on canary is fixed (see https://github.com/emberjs/ember-test-helpers/pull/1066), you could update `@ember/test-helpers` to 2.2.9+

But this scenario will probably still fails (see https://github.com/poteto/ember-cli-flash/pull/367, same import issue in this repo), maybe it is better to do the change on your PR so that we ensure ember-canary scenario will be ok?

**Comment by stukalin** [comment]
@ndekeister-us thanks for the comment! I'll try this one out

File: app/initializers/flash-messages.js [hunk 1]
```diff
@@ -1,20 +1,27 @@
...
-  const shouldShowDeprecation = !(injectionFactories && injectionFactories.length);
+  const shouldShowDeprecation = !(
+    injectionFactories && injectionFactories.length
+  );
 
   deprecate(INJECTION_FACTORIES_DEPRECATION_MESSAGE, shouldShowDeprecation, {
     id: 'ember-cli-flash.deprecate-injection-factories',
-    until: '2.0.0'
+    until: '2.0.0',
+    for: 'ember-cli-flash',
```
Comment (stukalin) [inline app/initializers/flash-messages.js:21]: Hope everyone's happy with that namespace

File: app/initializers/flash-messages.js [hunk 2]
```diff
@@ -1,20 +1,27 @@
...
+    since: {
+      available: '1.3.12',
```
Comment (stukalin) [inline app/initializers/flash-messages.js:23]: That was a long time ago, so I assume noone would complain if I put it wrong

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -15,7 +15,7 @@ You can contribute code, design (maybe 'themes' for different kinds of flash mes
 ## Linting
 
 * `yarn run lint:js`
-* `yarn run lint:js -- --fix`
+* `lint:js:fix`
```
Comment (ndekeister-us) [inline CONTRIBUTING.md:18]:
```suggestion
* `yarn lint:js:fix`
```

Or maybe replace l17+18 by 
```
`* `yarn lint:fix`
```
?

**Comment by stukalin** [comment]
@ndekeister-us fixed your comment. Would anyone else want to review/merge ther PR? @poteto ?

**Review by ndekeister-us — APPROVED** [review]

**Comment by poteto** [comment]
I don't work with Ember anymore so I don't think I can review this. Paging @sbatson5, @kategengler

**Comment by sbatson5** [comment]
> I don't work with Ember anymore so I don't think I can review this. Paging @sbatson5, @kategengler

I haven't worked in Ember for a couple years now, unfortunately.

**Comment by stukalin** [comment]
@kategengler may I ping you on this one then?

**Comment by Turbo87** [comment]
@poteto @sbatson5 if you both don't work with Ember anymore, would you be open to transfering the repo to https://github.com/adopted-ember-addons/ ?

instructions for that are available in https://github.com/adopted-ember-addons/program-guidelines#getting-your-addon-adopted

**Comment by sbatson5** [comment]
> @poteto @sbatson5 if you both don't work with Ember anymore, would you be open to transfering the repo to https://github.com/adopted-ember-addons/ ?
> 
> instructions for that are available in https://github.com/adopted-ember-addons/program-guidelines#getting-your-addon-adopted

I'm just a maintainer and don't have the rights to transfer this repo. I support the decision, however!

**Comment by poteto** [comment]
@Turbo87 Yeah I think that's the best option. I'll reach out to get this transferred.

**Review by rahulk94 — APPROVED** [review]
I've flicked through this PR and looks pretty standard / safe imo. Adding a review since it looks like the maintainers aren't super clear atm.

**Comment by stukalin** [comment]
@rahulk94 thanks a lot! Anyone with the "i am able to press the merge button" permission? @sbatson5 @poteto would you mind? (Sorry for bothering you each time guys)

**Comment by stukalin** [comment]
@poteto @Turbo87 any news on transferring the repo ?

**Comment by GreatWizard** [comment]
Thanks for you works here. The changes looks good to me too :)

This update is needed if we want to use ember v4... do you think you can merge and release it?

**Comment by stukalin** [comment]
@GreatWizard i don't have enough permissions unfortunately

**Comment by sbatson5** [comment]
👋 Sorry, for missing these pings. I can't cut a release if y'all are confident as we wait for the transfer to "adopted-ember-addons"

**Comment by rwwagner90** [comment]
@poteto @sbatson5  if you can transfer this repo to me, I am an owner on adopted-ember-addons and I can get this transferred there. I will also need you to please make me an owner on npm, I am rwwagner90 there as well.

**Pull request merged**